### PR TITLE
Rename all metadata according to the new convention

### PR DESCRIPTION
**Pull request by Luthaf**
This follows up to https://github.com/lab-cosmo/metatensor/pull/514, and changes most of the metadata names:

- `spherical_harmonics_l` => `o3_lambda`
- `species_xxx` => `xxx_type`
- `structure, center` => `system, atom`
- `spherical_harmonics_m` => `o3_mu`
- add a `o3_sigma` key to all spherical expansions

TODO:
- [x] release metatensor with the changes above and integrate them here
- [x] rename properties for power spectrum & CG

----
📚 Documentation preview 📚: https://rascaline--285.org.readthedocs.build/en/285/

### CODE REVIEW

**Review by frostedoyster — COMMENTED** [review]
Thanks!

File: python/rascaline-torch/rascaline/torch/utils.py [hunk 1]
```diff
@@ -42,6 +41,7 @@
 
 
 def is_labels(obj: Any):
+    return True
     return isinstance(obj, Labels)
```
[the remark this answers is not in the document]
Comment (Luthaf) [reply python/rascaline-torch/rascaline/torch/utils.py:45]: whoops, good catch!

File: python/rascaline/rascaline/calculators.py [hunk 2]
```diff
@@ -49,28 +49,12 @@ class NeighborList(CalculatorBase):
     of 0 (i.e. self pairs inside the original unit cell) are only included when using
     ``self_pairs=True``.
 
-    The ``quantity`` parameter determine what will be included in the output. It can
-    take one of three values:
+    This calculator produces a single property (``"distance"``) with three components
+    (``"pair_xyz"``) containing the x, y, and z component of the distance vector of the
```
[the remark this answers is not in the document]
Comment (frostedoyster) [reply python/rascaline/rascaline/calculators.py:53]: I think it's the same components, just in different contexts, so I think `xyz` would be ideal

**Review by frostedoyster — APPROVED** [review]

File: python/rascaline/rascaline/calculators.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (frostedoyster) [reply python/rascaline/rascaline/calculators.py:53]: I see the issue now. It doesn't look right to me, but let's go with pair_xyz

**Comment by frostedoyster** [comment]
```
      [ 20%] Building CXX object CMakeFiles/rascaline_torch.dir/src/system.cpp.o
      In file included from /home/filippo/rascaline/rascaline-torch/src/system.cpp:4:
      /home/filippo/rascaline/rascaline-torch/include/rascaline/torch/system.hpp:48:20: error: ‘const int32_t* rascaline_torch::SystemAdapter::types() const’ marked ‘override’, but does not override
         48 |     const int32_t* types() const override {
            |                    ^~~~~
      make[2]: *** [CMakeFiles/rascaline_torch.dir/build.make:76: CMakeFiles/rascaline_torch.dir/src/system.cpp.o] Error 1
      make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/rascaline_torch.dir/all] Error 2
      make: *** [Makefile:136: all] Error 2
```
I can't install it

**Comment by frostedoyster** [comment]
I can't seem to import `rascaline.torch`. This may or may not be related to the changes here, my environment might have issues
```
tests/utils/test_model_io.py:6: in <module>
    import rascaline.torch
../.local/lib/python3.8/site-packages/rascaline/torch/__init__.py:12: in <module>
    from . import utils  # noqa: E402, F401
../.local/lib/python3.8/site-packages/rascaline/torch/utils.py:87: in <module>
    spec.loader.exec_module(module)
../.local/lib/python3.8/site-packages/rascaline/utils/__init__.py:4: in <module>
    from .power_spectrum import PowerSpectrum  # noqa
../.local/lib/python3.8/site-packages/rascaline/utils/power_spectrum/__init__.py:1: in <module>
    from .calculator import PowerSpectrum
../.local/lib/python3.8/site-packages/rascaline/utils/power_spectrum/calculator.py:6: in <module>
    from ._classes import CalculatorBase, IntoSystem, Labels, TensorBlock, TensorMap
../.local/lib/python3.8/site-packages/rascaline/utils/power_spectrum/_classes.py:3: in <module>
    from ...calculator_base import CalculatorBase
E   ImportError: cannot import name 'CalculatorBase' from 'rascaline.torch.calculator_base' (/home/filippo/.local/lib/python3.8/site-packages/rascaline/torch/calculator_base.py)
```

**Comment by Luthaf** [comment]
Please run `./scripts/clean-python.sh` and then re-install everything!

**Pull request merged**